### PR TITLE
Use 'with' environment to enable signomials

### LINKED ISSUE #260: handle environments using with statements
as suggested by @bqpd, use of signomials would happen as follows:

```python
from gpkit import Variable
x = Variable('x')
y = Variable('y')
with gpkit.signomial_environment:
    sig = (x + y >= 1)
```

### PR DESCRIPTION
**Pull request by bqpd**
Closes #260

### CODE REVIEW

**Comment by bqpd** [comment]
restest this please

File: gpkit/__init__.py [hunk 1]
```diff
@@ -75,16 +75,26 @@ def enable_units():
...
+
+    Example
+    -------
+    >>> import gpkit
+    >>> x = gpkit.Variable("x")
+    >>> y = gpkit.Variable("y", 0.1)
+    >>> with enable_signomials():
+    >>>     constraints = [x >= 1-y]
+    >>> gpkit.Model(x, constraints).localsolve()
+    """
```
Comment (whoburg) [inline gpkit/__init__.py]: nice!!

Comment (whoburg) [inline gpkit/__init__.py]: is the idea of keeping the name the same to create backwards compatibility?

**Comment by whoburg** [comment]
This is so much cleaner!!

Only questions are
1. Should the class be called `EnableSignomials`?
2. Related to above, does the old method of calling `enable_signomials()` as a method work after we merge this, or will this break that?

 - [ ] Note merge conflict to fix...

**Comment by whoburg** [comment]
make sure this gets advertised in the 0.3.0 change log

**Comment by bqpd** [comment]
This will break backward compatibility, but it doesn't have to; it's just a different way of working. Should I add backwards compatibility in?

**Comment by whoburg** [comment]
If it's easy, I like the following:

1. change the class name of the `with` version to `EnableSignomials` (this follows standard class naming, right? or are there different conventions for environments?)
2. maintain the `enable_signomials` method (and `disable_signomials`) for compatibility, but add a deprecation warning explaining to use `with EnableSignomials` instead.

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
Changed the name.

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
@galbramc, this failed because of `FATAL: Could not checkout 57aa5bcd4259ca1915a9087dc547190d6097c31c`

**Comment by bqpd** [comment]
retest this please

**Comment by bqpd** [comment]
Yeah, it wasn't failing any tests, just failing to run. :-/

**Comment by bqpd** [comment]
I think this is ready to merge.

**Comment by galbramc** [comment]
So the error you got has something to do with a new commit added to the repo before all the different tests have checked out the current commit. Effectively, there is a newer command than the one that is supposed to be tested. I'm not sure why that results in this error though... I'm going to use pull request #329 to test a setting change in jenkins.

**Comment by galbramc** [comment]
So I don't know if the change fixed the problem, but it didn't break anything. I would need you to make a commit while the testing is running to see if the problem is fixed.

**Pull request merged**